File: app/src/main/java/com/zwstudio/logicpuzzlesandroid/puzzles/tapalike/domain/TapAlikeObject.java
package com.zwstudio.logicpuzzlesandroid.puzzles.tapalike.domain;

public abstract class TapAlikeObject {
    public abstract String objTypeAsString();
    public static TapAlikeObject objTypeFromString(String str) {
        switch (str) {
            case "marker":
                return new TapAlikeMarkerObject();
            case "wall":
                return new TapAlikeWallObject();
            default:
                return new TapAlikeEmptyObject();
        }
    }
}

File: app/src/main/java/com/zwstudio/logicpuzzlesandroid/puzzles/tapalike/TapAlikeObject.kt
package com.zwstudio.logicpuzzlesandroid.puzzles.tapalike

import com.zwstudio.logicpuzzlesandroid.common.domain.Position

enum class TapAlikeObject {
    Empty, Hint, Marker, Wall
}

class TapAlikeGameMove(val p: Position, var obj: TapAlikeObject = TapAlikeObject.Empty)
